collect_app/src/test/java/org/odk/collect/android/geo/MapFragmentFactoryImplTest.kt

```diff
@@ -0,0 +1,81 @@
+package org.odk.collect.android.geo
+
+import org.hamcrest.CoreMatchers.instanceOf
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Test
+import org.odk.collect.osmdroid.OsmDroidMapFragment
+import org.odk.collect.settings.InMemSettingsProvider
+import org.odk.collect.settings.keys.ProjectKeys
+
+class MapFragmentFactoryImplTest {
```

Comment:
I'm not seeing how the tests relate to the issue! I would expect something related to multiple projects. Or is writing a non-Espresso test for that too difficult?

These test the relationship between basemap source configs and the engine used by the map fragment, right? That does seem like a good thing to have coverage for that isn't covered elsewhere. Any reason not to include Mapbox?

Reply:
The problem was that `MapFragmentFactoryImpl` as a parameters received `Settings` and it was easy to make a mistake and pass incorrect ones (from a different project). Now it takes `SettingsProvider` so we do not need to test if we pass correct ones (there is one settings provider for all of them). We just need to verify that creating map fragments it takes settings that belong to the current project (from that `SettingsProvider`) what we do in the test because in both (tests and code base) we use `getUnprotectedSettings()` without any parameter (what returns settings for the current project)

> Any reason not to include Mapbox?

I've only tested `OSM` and `Google` because `Mapbox` classes do not exist without tokens so it's not possible to verify it.

Reply:
Thanks for the explanations!